code/modules/lootpanel/search_object.dm

```diff
@@ -0,0 +1,70 @@
+/**
+ * ## Search Object
+ * An object for content lists. Compacted item data.
+ */
+/datum/search_object
+	/// Client attached to the search_object
+	var/datum/weakref/client_ref
+	/// Weakref to the original object
+	var/datum/weakref/item_ref
+	/// Url to the image of the object
+	var/icon
+	/// Icon state, for inexpensive icons
+	var/icon_state
+	/// The name of the object
+	var/name
+	/// The typepath, used for concatenating the search results
+	var/path
+	/// The STRING reference of the object for indexing purposes
+	var/string_ref
+
+
+/datum/search_object/New(mob/user, atom/item)
+	. = ..()
+
+	client_ref = WEAKREF(user.client)
+	item_ref = WEAKREF(item)
+	name = item.name
+	path = isobj(item) && item.type
```

Comment:
what is item.type doing here

Reply:
i'm using the type path to group similar items in the UI

Reply:
so path is either FALSE or a typepath?
do you mind using an if block here instead I don't like the mixing of types (I know this is a lot like || but it's new and new things are scary)